setup_ffe.sh

```diff
@@ -0,0 +1,244 @@
+rm -f my-libdatadog-build/arm64-darwin-24/include/datadog/datadog_ffe.h.bak*
+
+# Remove duplicates from library-config.h too
+echo "Removing duplicate types from library-config.h..."
+sed -i.bak1 '/typedef struct ddog_Vec_U8 {/,/} ddog_Vec_U8;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h
+sed -i.bak2 '/typedef struct ddog_Error {/,/} ddog_Error;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h
+sed -i.bak3 '/typedef struct ddog_Slice_CChar {/,/} ddog_Slice_CChar;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h
+sed -i.bak4 '/typedef struct ddog_Slice_CChar ddog_CharSlice;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h
+rm -f my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h.bak*
+
+# Create minimal stub headers for libraries we're linking but don't actively use
+echo "Creating minimal stub headers for unused linked libraries..."
+
+# profiling.h - minimal stub  
+cat > my-libdatadog-build/arm64-darwin-24/include/datadog/profiling.h << 'EOF'
+#ifndef DDOG_PROFILING_H
+#define DDOG_PROFILING_H
+
+#pragma once
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "common.h"
+
+// Minimal declarations for profiling functionality
+// Ruby bindings don't need full profiling API, just enough to link
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+
+// Stub function declarations - can be extended as needed
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif  // __cplusplus
+
+#endif  /* DDOG_PROFILING_H */
+EOF
+
+# No more stub headers needed! We have real ones from cbindgen
+
+# Create pkg-config file for all FFI libraries
+echo "Creating pkg-config file..."
+CURRENT_DIR=$(pwd)
+cat > my-libdatadog-build/pkgconfig/datadog_profiling_with_rpath.pc << EOF
+prefix=${CURRENT_DIR}/my-libdatadog-build
+exec_prefix=\${prefix}
+libdir=\${exec_prefix}/arm64-darwin-24/lib
+includedir=\${prefix}/arm64-darwin-24/include
+
+Name: datadog_profiling_with_rpath
+Description: Datadog libdatadog library (with rpath) - Full FFI build
+Version: 22.1.0
+Libs: -L\${libdir} -ldatadog_ffe_ffi -ldatadog_crashtracker_ffi -lddsketch_ffi -ldatadog_library_config_ffi -ldatadog_profiling_ffi -Wl,-rpath,\${libdir}
+Cflags: -I\${includedir}
+EOF
+
+echo "✅ Step 2 completed: Build environment set up"
+
+# Step 3: Compile Ruby Extension
+echo "🔨 Step 3: Compiling Ruby extension..."
+
+# Set PKG_CONFIG_PATH to find our custom build
+export PKG_CONFIG_PATH="$(pwd)/my-libdatadog-build/pkgconfig:$PKG_CONFIG_PATH"
+echo "PKG_CONFIG_PATH set to: $PKG_CONFIG_PATH"
+
+# Compile the Ruby extension
+cd ext/libdatadog_api
+echo "Generating Makefile..."
+ruby extconf.rb
+echo "Compiling extension..."
+make
+echo "Installing extension..."
+cp libdatadog_api.*.bundle ../../lib/
+echo "Cleaning up build artifacts..."
+make clean
+rm -f Makefile
+cd ../..
```

Comment:
Minor: We have [rake-compiler](https://github.com/rake-compiler/rake-compiler) as a dependency, you can use `bundle exec rake compile:libdatadog_api.NAME_OF_THE_PLATFORM` to build the extension (You can use `bundle exec rake -T` to see all targets and which platform name you should use)

Reply:
Thanks for the tip! Updated in https://github.com/DataDog/dd-trace-rb/pull/5007/commits/e974e4e8734100a0e1929e90ad93952313550e67

Reply:
I've never looked into it, but we could probably get rake to generate an alias that doesn't include the platform name as the task suffix -- the suffix comes from `rake-compiler`.